### PR TITLE
WIP: R keras 2.2.5.0 fosscuda 2019b

### PR DESCRIPTION
**Pull request by edmondac**
Requires https://github.com/bear-rsg/easybuild-easyconfigs/pull/21 to be merged first

Includes fixes for R and deps on POWER9

For INC986978

`R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb`
* [x] Assigned to reviewer
* [ ] EL7-haswell
* [ ] EL7-cascadelake
* [ ] EL7-sandybridge
* [ ] Ubuntu16.04-haswell
* [ ] EL7-power9
* [ ] Created gitlab issues for testing and examples

`R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb`
* [ ] EL7-haswell
* [ ] EL7-power9
* [ ] Created gitlab issues for testing and examples

### CODE REVIEW

File: easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+configopts = '-DCMAKE_BUILD_TYPE=Release'
+
+exts_defaultclass = 'PythonPackage'
+exts_default_options = {
+    'download_dep_fail': True,
+    'use_pip': True,
+}
+exts_list = [
+    ('numpy', '1.17.4', {
```
Comment (branfosj) [inline easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb]: Does this really need `numpy` 1.17.4? SciPy-bundle has 1.17.3...

File: easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+description = "Library to manipulate tensors on the GPU."
+
+toolchain = {'name': 'fosscuda', 'version': '2019b'}
+
+source_urls = ['https://github.com/Theano/libgpuarray/releases/download/v%(version)s/']
+sources = [SOURCE_TAR_GZ]
+checksums = ['887b6433a30282cb002117da89b05812c770fd9469f93950ff3866ddd02bfc64']
+
+# numpy 1.17.4 says 'Python version >= 3.5 required'
+multi_deps = {'Python': ['3.7.4']}
```
Comment (branfosj) [inline easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb]: If there is only one version of Python then this should become a dep and the filename renamed accordingly.

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('R', '3.6.2'),
+    ('Keras', '2.3.1', '-Python-%(pyver)s'),
+]
+
+exts_defaultclass = 'RPackage'
+exts_default_options = {
+    'source_urls': ['https://cran.r-project.org/src/contrib/'],
```
Comment (branfosj) [inline easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb]: Should be:
``` 
   'source_urls': [
        'https://cran.r-project.org/src/contrib/Archive/%(name)s',  # package archive
        'https://cran.r-project.org/src/contrib/',  # current version of packages
        'https://cran.freestatistics.org/src/contrib',  # mirror alternative for current packages
    ],
```

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+dependencies = [
+    ('Python', '3.7.4'),
+    ('R', '3.6.2'),
+    ('Keras', '2.3.1', '-Python-%(pyver)s'),
+]
+
+exts_defaultclass = 'RPackage'
+exts_default_options = {
+    'source_urls': ['https://cran.r-project.org/src/contrib/'],
```
Comment (branfosj) [inline easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb]: Should be:
``` 
   'source_urls': [
        'https://cran.r-project.org/src/contrib/Archive/%(name)s',  # package archive
        'https://cran.r-project.org/src/contrib/',  # current version of packages
        'https://cran.freestatistics.org/src/contrib',  # mirror alternative for current packages
    ],
```

File: easybuild/easyconfigs/t/Theano/Theano-1.0.4-fosscuda-2019b.eb [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+description = """Theano is a Python library that allows you to define, optimize,
+and evaluate mathematical expressions involving multi-dimensional arrays efficiently."""
+
+toolchain = {'name': 'fosscuda', 'version': '2019b'}
+
+source_urls = [PYPI_SOURCE]
+sources = [SOURCE_TAR_GZ]
+checksums = ['35c9bbef56b61ffa299265a42a4e8f8cb5a07b2997dabaef0f8830b397086913']
+
+# numpy version in libgpuarray requires Python 3
```
Comment (branfosj) [inline easybuild/easyconfigs/t/Theano/Theano-1.0.4-fosscuda-2019b.eb]: If there is only one version of Python then this should become a dep and the filename renamed accordingly.

File: easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb (hunk 1 above)
Comment (edmondac) [reply easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb]: I've included SciPy-bundle instead

File: easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb (hunk 2 above)
Comment (edmondac) [reply easybuild/easyconfigs/l/libgpuarray/libgpuarray-0.7.6-fosscuda-2019b.eb]: Done

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb (hunk 3 above)
Comment (edmondac) [reply easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-foss-2019b-Python-3.7.4-R-3.6.2.eb]: Done

File: easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb (hunk 4 above)
Comment (edmondac) [reply easybuild/easyconfigs/r/R-keras/R-keras-2.2.5.0-fosscuda-2019b-Python-3.7.4-R-3.6.2.eb]: Done

File: easybuild/easyconfigs/t/Theano/Theano-1.0.4-fosscuda-2019b.eb (hunk 5 above)
Comment (edmondac) [reply easybuild/easyconfigs/t/Theano/Theano-1.0.4-fosscuda-2019b.eb]: Done

**Pull request merged**